components/src/main/java/org/folio/rest/workflow/model/resolver/DeserializeAsJsonResolver.java

```
@@ -0,0 +1,41 @@
+package org.folio.rest.workflow.model.resolver;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+import com.fasterxml.jackson.databind.DatabindContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.jsontype.impl.TypeIdResolverBase;
+
+public class DeserializeAsJsonResolver extends TypeIdResolverBase {
+
+  @Override
+  public String idFromValue(Object value) {
+    return value.getClass().getSimpleName();
+  }
+
+  @Override
+  public String idFromValueAndType(Object value, Class<?> suggestedType) {
+    return idFromValue(value);
+  }
+
+  @Override
+  public Id getMechanism() {
+    return Id.SIMPLE_NAME;
+  }
+
+  @Override
+  public JavaType typeFromId(DatabindContext context, String id) {
+    if (id.length() > 0) {
+      try {
+        Class<?> type = Class.forName("org.folio.rest.workflow.model." + id);
+        if (type != null) {
+          return context.getTypeFactory().constructType(type);
+        }
+      } catch (ClassNotFoundException e) {
+        e.printStackTrace();
```

I think this should be handled better. Should we consider throwing a custom runtime exception and avoid returning null?

---

NULL is a perfectly reasonable return.
- https://www.baeldung.com/jackson-advanced-annotations

---

Is it not reasonable to throw an exception as well? I think it would be more predictable throwing a custom runtime exception rather than having some flow of code handle a null response here.

https://johnhunsley.blogspot.com/2016/12/jackson-type-id-mapping.html

---

Why not throw a [JsonMappingException](https://fasterxml.github.io/jackson-databind/javadoc/2.8/com/fasterxml/jackson/databind/JsonMappingException.html)?